File: backend/src/main/scala/merchant/api/UpdateMerchantProfileApi.scala
package merchant.api

import domain.shared.given

import cats.effect.IO
import domain.merchant.UpdateMerchantProfileRequest
import domain.shared.{DeliveryAppState, UserRole}
import merchant.app.*
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val updateMerchantProfileApi: FixedMethodApi0[DeliveryAppState] =
  jsonPostApi0[UpdateMerchantProfileRequest, DeliveryAppState](
    routeSegment("api"),
    routeSegment("delivery"),
    routeSegment("merchant-profile"),
  )

File: frontend/src/apis/merchant/UpdateMerchantProfileApi.ts
import type { DeliveryAppState, UpdateMerchantProfileRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { defineJsonPostApi0, routeSegment } from '@/system/api/TypedApiDefinitions'

export const updateMerchantProfileApiDefinition = defineJsonPostApi0<UpdateMerchantProfileRequest, DeliveryAppState>([
    routeSegment('api'),
    routeSegment('delivery'),
    routeSegment('merchant-profile'),
  ])

export function updateMerchantProfile(payload: UpdateMerchantProfileRequest) {
  return postNormalizedDeliveryState(updateMerchantProfileApiDefinition, payload)
}
